wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -379,27 +382,49 @@ public void checkout(CheckoutParams params) throws GitException {
       if (startPoint != null && trackBranch != null) {
         throw new GitException("Start point and track branch can not be used together.");
       }
-      if (params.isCreateNew() && name == null) {
+      if (params.isCreateNew() && branchName == null) {
         throw new GitException("Branch name must be set when createNew equals to true.");
       }
       if (startPoint != null) {
         checkoutCommand.setStartPoint(startPoint);
       }
       if (params.isCreateNew()) {
         checkoutCommand.setCreateBranch(true);
-        checkoutCommand.setName(name);
-      } else if (name != null) {
-        checkoutCommand.setName(name);
+        checkoutCommand.setName(branchName);
+      } else if (branchName != null) {
+        checkoutCommand.setName(branchName);
         List<String> localBranches =
             branchList(LIST_LOCAL)
                 .stream()
                 .map(Branch::getDisplayName)
                 .collect(Collectors.toList());
-        if (!localBranches.contains(name)) {
+        if (!localBranches.contains(branchName)) {
+          List<Branch> branches = branchList(LIST_REMOTE);
+          List<String> remotes =
+              repository
+                  .getConfig()
+                  .getSubsections(ConfigConstants.CONFIG_KEY_REMOTE)
+                  .stream()
+                  .filter(
+                      remote ->
+                          branches
+                              .stream()
+                              .anyMatch(
+                                  branch ->
+                                      ("refs/remotes/" + remote + "/" + branchName)
+                                          .equals(branch.getName())))
+                  .collect(Collectors.toList());
+          if (remotes.size() > 1 || remotes.size() == 0) {
+            throw new GitException(String.format(ERROR_CHECKOUT_BRANCH_NAME_NOT_FOUND, branchName));
+          }
+
           Optional<Branch> remoteBranch =
-              branchList(LIST_REMOTE)
+              branches
```

Comment:
Accordingly to the all entries in the `branches` are equal to the `branch.getName()`
```
("refs/remotes/" + remote + "/" + branchName)
 +                                          .equals(branch.getName())
```